<launch>
  
  <!-- launch file for processing pointclouds from aubo_control main_LIDAR.launch -->
  <env name="LIBGL_ALWAYS_SOFTWARE" value="1"/>

  <!-- read arg value load the parameters from 'main' config file-->
  <arg name="config" default="aubo_i10_target_source" />
  <param name="param" value="$(arg config)"/> 
  <rosparam file="$(find seam_detection)/config/$(arg config).yaml" /> 
 
  <!-- setup the transformations and frames for the system-->
  <node
    pkg="seam_detection"
    name="seam_detection_tf"
    type="seam_detection_tf"
    output="screen">
  </node>


  <!-- set params to save target -->
  <rosparam param="saving_target">True</rosparam>
  <rosparam param="saving_source">False</rosparam>

  <rosparam param="filter_cloud/output_file">"pcd_images/aubo_i10/aubo_cloud_out_target.pcd"</rosparam>
    
  <!-- get the target cloud published by the aubo robot arm -->
  <node                   
    pkg="seam_detection"          
    type="get_target"
    name="get_target"
    output="screen">
  </node>

  <!-- save the target cloud published by the aubo robot arm --> 
  <node                   
    pkg="seam_detection"          
    type="filter_target"
    name="filter_target"   
    output="screen">
  </node>


  <!-- set params to save source -->
  <rosparam param="saving_target">False</rosparam>
  <rosparam param="saving_source">True</rosparam>
  <rosparam param="filter_cloud/output_file">"pcd_images/aubo_i10/aubo_cloud_out_source.pcd"</rosparam>
    
  <!-- get the cloud published by the aubo robot arm -->
  <node                   
    pkg="seam_detection"             
    type="get_source"
    name="get_source"
    output="screen">
    args
  </node>

  <!-- save the cloud published by the aubo robot arm --> 
  <node                   
    pkg="seam_detection"          
    type="filter_source"   
    name="filter_source"
    output="screen">
  </node>


  <!-- processing of lidar data for registration -->
  <node
    pkg="seam_detection"
    name="register_clouds"
    type="register_clouds"
    output="screen">
  </node>

  <!-- start RVIZ --> 
  <node 
    name="rviz20"
    pkg="rviz"
    type="rviz"
    args="-d $(find seam_detection)/rviz/aubo_i10_lidar.rviz">
  </node>

  <!-- load aubo i10 model from URDF with xacro -->
  <param name="robot_description" command="xacro $(find aubo_description)/urdf/aubo_i10_robot.xacro" />

</launch>

